### DISCUSSION TITLE
Object fall down into the floor and float up slowly.

### DISCUSSION THREAD

**Discussion by feixinyv** [jrouwe/JoltPhysics#221] [Q&A]
I integrate Jolt into UnrealEngine 5, right now I only finished basic parts, things like physics material lefted untouched. 
The vedio below shows two cube, one as floor, and one as a stick up in the air, when the stick falls down, it will insert into the floor and float up slowly.
https://user-images.githubusercontent.com/11873408/187613185-4d9d3888-f418-406c-bc43-623592c1f3ca.mp4
UE5 use coordinate system with Z-up, the stick is about 6 meters from the ground. Update frenquency is 30Hz.
I failed to reproduce it in the Samples project of this repository.  What could possibly cause the problem? 
Thank you for your attention to this matter.

**Comment by jrouwe** [comment]
I would first try increasing the update frequency to 60Hz (you can do this by setting inCollisionSteps = 2 in the PhysicsSystem::Update function).

What kind of units are you passing into Jolt? I think Unreal uses cm while Jolt uses m. Are you converting these? (there are a number of tolerance values that won't work well if you change the units)

**Reply by feixinyv** [reply]
I have convert unit to M, I tried to increase the whole update frenquency to 60Hz, it will decrease the probability of insertion and the extent of insertion, but the situation will still happen.

**Reply by jrouwe** [reply]
It's really hard to say anything sensible without some sort of code/example. Have you tried setting the motion quality of the body to LinearCast?

**Reply by feixinyv** [reply]
I think I could upload the intergated version in UE5 later, would you help me to take a look at this problem in Unreal Engine？

**Reply by jrouwe** [reply]
I can take a look at the code. Don't have much experience with UE, so I don't know if I'll be able to actually test it.

**Comment by jrouwe** [comment]
B.t.w. very cool project! Are you sharing what you're doing on github?

**Reply by feixinyv** [reply]
Not yet，I first integrate it as a Game plugin which will be easily sharing. But now we decide to eventally replace UE's physics engine if all goes well, which will be difficult to share.
